src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/TransactionService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.limitedpartnershipsapi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.api.sdk.ApiClientService;
+import uk.gov.companieshouse.limitedpartnershipsapi.exception.ServiceException;
+import uk.gov.companieshouse.limitedpartnershipsapi.utils.ApiLogger;
+
+import java.io.IOException;
+
+import static uk.gov.companieshouse.limitedpartnershipsapi.utils.Constants.TRANSACTIONS_PRIVATE_API_URI_PREFIX;
+
+@Service
+public class TransactionService {
+
+    @Autowired
+    private final ApiClientService apiClientService;
+
+    @Autowired
+    public TransactionService(ApiClientService apiClientService) {
+        this.apiClientService = apiClientService;
+    }
+
+    public void updateTransaction(Transaction transaction, String loggingContext) throws ServiceException {
+        try {
+            var uri = TRANSACTIONS_PRIVATE_API_URI_PREFIX + transaction.getId();
+
+            // The internal API key client is used here as the transaction service will potentially call back into the
+            // LP API (e.g. to get the costs, if a costs end-point has already been set on the transaction) and those
+            // calls cannot be made with a user OAuth token
+            var response = apiClientService.getInternalApiClient()
+                    .privateTransaction().patch(uri, transaction).execute();
+
+            if (response.getStatusCode() != HttpStatus.NO_CONTENT.value()) {
```

Comment:
switch the sides of the comparison to avoid npe, 
`if (HttpStatus.NO_CONTENT.value() != response.getStatusCode() )`

Reply:
sorry, would still get a npe when switched if response was null

Reply:
I don't think this is needed.

Reason being that if response _is_ null we would get a (runtime) `NullPointerException` on line 37 which would ultimately end up with a 500 internal error response to the client (the full stack trace would still get logged in the API by the controller code, so no details lost). Checking for `null` here would add a few more lines of code/logic but ultimately end up with the same outcome.